app/controllers/pages_controller.rb

```diff
@@ -0,0 +1,44 @@
+class PagesController < ApplicationController
+before_action :set_page, only: [:show, :edit, :update, :destroy]
+
+    def index
+        @pages = Page.all
+    end
+
+    def show
+    end
+
+    def new
+        @page = Page.new
+    end
+
+    def create
+        @page = Page.new(page_params)
+        @page.save
+        redirect_to @page
+    end
+
+    def edit
+        set_page
```

Comment:
You don't need this line; `set_page` is already called on `edit` because of what you passed into `before_action` above.